src/mem/largealloc.h

```diff
@@ -145,32 +145,39 @@ namespace snmalloc
       {
         FlagLock f(lock);
 
-        auto aligned_bump = bits::align_up(bump, alignment);
-        if ((aligned_bump - bump) > remaining)
+        if constexpr (alignment != 0)
         {
-          new_block();
-        }
-        else
-        {
-          remaining -= aligned_bump - bump;
-          bump = aligned_bump;
+          uint8_t* aligned_bump = pointer_align_up<alignment, uint8_t>(bump);

```

Comment:
Should this be `char` rather than `uint8_t` based on @davidchisnall's comments for the earlier PR?

Reply:
Probably so; fixed.